ui/src/Main.elm

```diff
@@ -310,11 +310,13 @@ view model =
 
             menu =
                 if loggedIn then
-                    [ ButtonGroup.linkButton [ Button.primary, Button.onClick Version, buttonAttrs ] [ i [ class "fas fa-chart-bar" ] [], text " Dashboard" ]
-                    , ButtonGroup.linkButton [ Button.primary, Button.onClick Things, buttonAttrs ] [ i [ class "fas fa-sitemap" ] [], text " Things" ]
-                    , ButtonGroup.linkButton [ Button.primary, Button.onClick Channels, buttonAttrs ] [ i [ class "fas fa-broadcast-tower" ] [], text " Channels" ]
-                    , ButtonGroup.linkButton [ Button.primary, Button.onClick Connection, buttonAttrs ] [ i [ class "fas fa-plug" ] [], text " Connection" ]
-                    , ButtonGroup.linkButton [ Button.primary, Button.onClick Messages, buttonAttrs ] [ i [ class "far fa-paper-plane" ] [], text " Messages" ]
+                    [ ul [ class "nav-pills flex-column nav" ]
+                        [ menuItem "Dashboard" Version "fas fa-chart-bar" (model.view == "dashboard")
+                        , menuItem "Things" Things "fas fa-sitemap" (model.view == "things")
+                        , menuItem "Channels" Channels "fas fa-broadcast-tower" (model.view == "channels")
+                        , menuItem "Connection" Connection "fas fa-plug" (model.view == "connection")
+                        , menuItem "Messages" Messages "far fa-paper-plane" (model.view == "messages")
+                        ]
```

Comment:
Ok for `ul`. Please replace `as fa-chart-bar` (magic string) and similar by some constant pulled from some module.